src/ProductsSpotter/ProductsSpotter.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\ProductsSpotter;
+
+
+use App\ProductImportRulesChecker\ProductImportRulesChecker;
+
+/**
+ * This class spots products parameters;
+ * Removes products with repeatable code;
+ * Applies the third rule to products.
+ * Also this class can show spot errors.
+ */
+
+class ProductsSpotter
+{
+    private const DATE_DISCONTINUED="dtmDiscontinued";
+    private const COST="Cost in GBP";
+    private const STOCK="Stock";
+    private const IS_DISCONTINUED="Discontinued";
+    private const CORRECT_ANSWER="yes";
+    private const PRODUCT_CODE="Product Code";
+    private ProductImportRulesChecker $productRuleChecker;
+    private array $errors;
+    private array $products;
+
+    public function __construct(array $products,array $errors)
+    {
+        $this->products=$products;
+        $this->productRuleChecker=new ProductImportRulesChecker();
+        $this->errors=$errors;
+    }
+
+    public function spotProducts():array
+    {
+        $this->spotProductsProductCode();
+        $this->spotProductsParams();
+        $this->spotProductsThirdRule();
+        $spottedProducts=[];
+        $i=0;
+        foreach ($this->products as $product){
+            if($this->productRuleChecker->check($product)){
+                $spottedProducts[]=$product;
+            }
+            else{
+                if(isset($this->errors[$product[self::PRODUCT_CODE]])) {
+                    $this->errors[$product[self::PRODUCT_CODE]] .= $this->productRuleChecker->getErrorMessage();
+                }
+                else{
+                    $this->errors[$product[self::PRODUCT_CODE]] = $this->productRuleChecker->getErrorMessage();
+                }
+            }
+            $this->productRuleChecker->resetErrorMessage();
+        }
+        return $spottedProducts;
+    }
+
+    //converts string cost to float and string stock to int.
+    private function spotProductsParams():void
+    {
+        foreach ($this->products as &$product){
+
+            $product[self::COST]=floatval($product[self::COST]);
+            $product[self::STOCK]=intval($product[self::STOCK]);
+            if(strcmp($product[self::IS_DISCONTINUED],self::CORRECT_ANSWER)!==0){
```

Comment:
`if ($product[self::IS_DISCONTINUED] !== self::CORRECT_ANSWER) {`
Same for any other strcmp in your code. Do not use it.